File: hackerrank/day17/main.js
'use strict';

const readline = require('readline');

class Calculator {

    power(n, p) {
        if (n < 0 || p < 0) {
            throw new Error("n and p should be non-negative");
        }

        return n ** p;
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const calculator = new Calculator();

rl.question("¿Cuántos casos desea probar? ", (respuesta) => {

    const cantidad = parseInt(respuesta);
    let contador = 0;

    function siguienteCaso() {

        if (contador === cantidad) {
            rl.close();
            return;
        }

        rl.question(`Caso #${contador + 1}: `, (entrada) => {

            const [n, p] = entrada.split(' ').map(Number);

            try {
                const resultado = calculator.power(n, p);
                console.log("Resultado:", resultado);
            }
            catch (error) {
                console.log(error.message);
            }

            contador++;
            siguienteCaso();
        });
    }

    siguienteCaso();
});

File: hackerrank/day17/main.cs
using System;

class Calculator
{
    public int Power(int n, int p)
    {
        if (n < 0 || p < 0)
        {
            throw new Exception("n and p should be non-negative");
        }

        int resultado = 1;

        for (int i = 0; i < p; i++)
        {
            resultado *= n;
        }

        return resultado;
    }
}

class Program
{
    static void Main(string[] args)
    {
        Calculator calculator = new Calculator();

        Console.Write("¿Cuántos casos desea probar? ");
        int cantidad = int.Parse(Console.ReadLine());

        for (int i = 0; i < cantidad; i++)
        {
            Console.Write($"Caso #{i + 1}: ");

            string[] datos = Console.ReadLine().Split(' ');

            int n = int.Parse(datos[0]);
            int p = int.Parse(datos[1]);

            try
            {
                int resultado = calculator.Power(n, p);
                Console.WriteLine("Resultado: " + resultado);
            }
            catch (Exception error)
            {
                Console.WriteLine(error.Message);
            }
        }
    }
}